### PR TITLE
added strace exclude filter feature 

### PR DESCRIPTION
**Pull request by salsal97**

### CODE REVIEW

File: tools/myst/host/exec_linux.c [hunk 1]
```diff
@@ -82,6 +82,27 @@ Options:\n\
                             it encounters an unimplemented syscall\n\
                             'true' implies the syscall would not terminate\n\
                             and instead return ENOSYS.\n\
+    --strace\n\
+                         -- Use this option to display the strace of the execution\n\
```
Comment (vtikoo) [inline tools/myst/host/exec_linux.c:86]:
```suggestion
                         -- Use this option to display the system call traces of the execution\n\
```

File: tools/myst/host/exec_linux.c [hunk 2]
```diff
@@ -82,6 +82,27 @@ Options:\n\
...
+    --strace-failing\n\
+                         -- When specified, all syscalls that fail will be logged.\n\
+                           Other syscalls will not be logged, unless specified via \n\
+                           filter (see below).\n\
+                           Set a breakpoing in _strace_failure_hook to stop \n\
```
Comment (vtikoo) [inline tools/myst/host/exec_linux.c:91]: Is this a debugging suggestion?

**Review by vtikoo — APPROVED** [review]
Code changes look good! Made some comments about documentation.

File: tools/myst/host/exec_linux.c (hunk 2 above)
Comment (salsal97) [reply tools/myst/host/exec_linux.c:91]: Yes I think so, picked it up from the PR that added this feature

Comment (vtikoo) [reply tools/myst/host/exec_linux.c:91]: We can also suggest using syscall number as a condition in the breakpoint.
Also there is a typo in breakpoing.

File: doc/user-getting-started.md [hunk 3]
```diff
@@ -73,12 +73,29 @@ Options:
...
+                         -- When specified, all syscalls that fail will be logged.
+                           Other syscalls will not be logged, unless specified via filter (see below).
+                           Set a breakpoing in _strace_failure_hook to stop execution whenever a
+                           syscall fails. Use breakpoint conditions to control the behavior of the breakpoint.
+    --strace-filter "SYS_name1:SYS_name2:..."
+                         -- Specify the set of syscalls to be traced. When filters are specified, only those
+                           syscalls specified in the filter will be traced, in addition to failing syscalls if
+                           specified as described above.
+                           E.g: To trace open and mprotect syscalls, specify
+                           --strace-filter "SYS_open:SYS_mprotect"
```
Comment (asvrada) [inline doc/user-getting-started.md:94]: (Very minor) The indentation seems to miss by 1 space

File: tools/myst/host/exec.c [hunk 4]
```diff
@@ -385,6 +385,27 @@ Options:\n\
...
+                           Set a breakpoing in _strace_failure_hook to stop \n\
+                           execution whenever a syscall fails. Use breakpoint\n\
+                           conditions to control the behavior of the breakpoint.\n\
+    --strace-filter 'SYS_name1:SYS_name2:...'\n\
+                         -- Specify the set of syscalls to be traced. When filters\n\
+                            are specified, only those syscalls specified in the \n\
+                            filter will be traced, in addition to failing syscalls if\n\
+                           specified as described above.\n\
+                           E.g: To trace open and mprotect syscalls, specify\n\
+                           --strace-filter 'SYS_open:SYS_mprotect'\n\
```
Comment (asvrada) [inline tools/myst/host/exec.c:403]: Same indentation issue

**Review by asvrada — APPROVED** [review]

File: tools/myst/host/exec_linux.c (hunk 1 above)
Comment (salsal97) [reply tools/myst/host/exec_linux.c]: Added this wording

File: tools/myst/host/exec_linux.c (hunk 2 above)
Comment (salsal97) [reply tools/myst/host/exec_linux.c]: Added that wording

File: doc/user-getting-started.md (hunk 3 above)
Comment (salsal97) [reply doc/user-getting-started.md]: Thanks for pointing it out, I've fixed it for this line and other parts of this menu

**Pull request merged**